tools/outputapi/ENBinaryOutDiff.py

```diff
@@ -1 +0,0 @@
    else:
        raise Exception("only implemented on mac/linux")
    
    BinFile1 = OutputObject(dllLoc)
    BinFile1.OpenOutputFile(args[1])
    BinFile1.get_NetSize()
    BinFile1.get_Times()

    BinFile2 = OutputObject(dllLoc)
    BinFile2.OpenOutputFile(args[2])

    ####ENR_NodeAttribute;
    ##ENR_demand   = 0
    ##ENR_head     = 1
    ##ENR_pressure = 2
    ##ENR_quality  = 3
    NumberOfNodeAttr = 4

    ####ENR_LinkAttribute;
    ##ENR_flow         = 0
    ##ENR_velocity     = 1
    ##ENR_headloss     = 2
    ##ENR_avgQuality   = 3
    ##ENR_status       = 4
    ##ENR_setting      = 5
    ##ENR_rxRate       = 6
    ##ENT_frctnFctr    = 7
    NumberOfLinkAttr = 8

    NumberOfPeriods = BinFile1.numPeriods

    # Set Tolerances for each attribute
    # demand, head, pressure, quality
    NodeAttributeTolerances = [1e-6, 1e-6, 1e-6, 1e-6]
    # flow, velocity, headloss, avgQuality, status, setting, rxRate, frctnFctr
    LinkAttributeTolerances = [1e-6, 1e-6, 1e-6, 1e-6, 1e-6, 1e-6, 1e-6, 1e-6]

    #Compare Node Attributes
    for nodeAttrInd in range(NumberOfNodeAttr):
        for TSind in range(NumberOfPeriods):
            #Get 1 attribute for all nodes at time t
            NodeAttributeOut1 = BinFile1.get_NodeAttribute(nodeAttrInd, TSind)
            #Get 1 attribute for all nodes at time t
            NodeAttributeOut2 = BinFile2.get_NodeAttribute(nodeAttrInd, TSind)
            for Nodeind, NodeAttrVal in enumerate(NodeAttributeOut1):
                if abs(NodeAttrVal - NodeAttributeOut2[Nodeind]) > 0:
                    diff = abs(NodeAttrVal - NodeAttributeOut2[Nodeind] )
                    if diff > NodeAttributeTolerances[nodeAttrInd]:
                    	print("Node {0} attribute {1} exceeded tolerance. value 1 = {2} :: value 2 = {3} :: delta = {4}".format(Nodeind, nodeAttrInd, NodeAttrVal,  NodeAttributeOut2[Nodeind], diff) )
                        return False

    #Compare Link Attributes                
    for linkAttrInd in range(NumberOfLinkAttr):
        for TSind in range(NumberOfPeriods):
            #Get 1 attribute for all links at time t
            LinkAttributeOut1 = BinFile1.get_LinkAttribute(linkAttrInd, TSind)
            #Get 1 attribute for all links at time t
            LinkAttributeOut2 = BinFile2.get_LinkAttribute(linkAttrInd, TSind)
            for linkind, LinkAttrVal in enumerate(LinkAttributeOut1):
                if abs(LinkAttrVal - LinkAttributeOut2[linkind]) > 0:
                    diff = abs(LinkAttrVal - LinkAttributeOut2[linkind] )
                    if diff > LinkAttributeTolerances[linkAttrInd]:
                    	print("Link {0} attribute {1} exceeded tolerance. value 1 = {2} :: value 2 = {3} :: delta = {4}".format(linkind, linkAttrInd, LinkAttrVal,  LinkAttributeOut2[linkind], diff) )
                        return False
    return True


if __name__ == '__main__':
    if(BinCompare(sys.argv)):
    	sys.exit(0)
    else:
        sys.exit(1)

```

Comment:
what's with the line breaks?

Reply:
i know this file is removed. just curious

Reply:
The line breaks and line wrap when the file was displayed were all messed up. I have no idea why.